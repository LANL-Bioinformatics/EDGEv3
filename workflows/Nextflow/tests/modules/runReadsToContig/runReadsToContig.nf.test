nextflow_workflow {

    name "Test Workflow READSTOCONTIGS"
    script "modules/runReadsToContig/runReadsToContig.nf"
    workflow "READSTOCONTIGS"

    test("BWA mem") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testR2C"
                }
                r2c {
                    r2c_aligner = "bwa"
                    extractUnmapped = true
                }
            }
            workflow {
                """
                input[0] = params.r2c.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                input[3] = file("${projectDir}/test_data/contigs.fa")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success}
            ) 
        }

    }

    test("Bowtie2") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testR2C"
                }
                r2c {
                    r2c_aligner = "bowtie2"
                    extractUnmapped = true
                }
            }
            workflow {
                """
                input[0] = params.r2c.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                input[3] = file("${projectDir}/test_data/contigs.fa")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success}
            ) 
        }

    }

    test("minimap2") {

        when {
            params {
                shared {
                    outDir = "./testing_output"
                    projName = "testR2C"
                }
                r2c {
                    r2c_aligner = "minimap2"
                    extractUnmapped = true
                }
            }
            workflow {
                """
                input[0] = params.r2c.plus(params.shared)
                input[1] = [file("${projectDir}/test_data/Ecoli_10x.1.fastq"),file("${projectDir}/test_data/Ecoli_10x.2.fastq")]
                input[2] = file("${projectDir}/nf_assets/NO_FILE2")
                input[3] = file("${projectDir}/test_data/contigs.fa")
                """
            }
        }

        then {
            assertAll(
                {assert workflow.success}
            ) 
        }

    }


}
