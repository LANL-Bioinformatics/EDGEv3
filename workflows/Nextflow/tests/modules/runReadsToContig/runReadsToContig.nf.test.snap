{
    "BWA mem": {
        "content": [
            "Final_contigs.fasta:md5,cfb6bc7595556231e115e5c9ad57f20d",
            "readsToContigs_coverage.table.json:md5,ea221b314df455a84177b669edb3e5b7",
            "readsToContigs.alnstats.txt:md5,259f501e745b79cb92b69e335d01cac7"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-17T13:52:25.552513444"
    }
}